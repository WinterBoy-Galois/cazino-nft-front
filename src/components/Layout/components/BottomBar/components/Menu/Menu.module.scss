@import '../../../../../../styles/shared';

.container {
  display: flex;
  margin: 0 0.5rem;
}

.item,
.item_open,
.item_menu,
.item_menu_open {
  width: 40px;
  padding: 5px;
  cursor: pointer;
  position: relative;
  margin-right: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.item {
  @include media-breakpoint-up(md) {
    width: 53px;
    padding: 12px 6px;
  }
  &:last-child {
    margin-right: 0;
  }

  &__icon {
    width: 35px;
    height: 35px;
    fill: $orange;
  }

  &__badge {
    width: 15px;
    height: 15px;
    position: absolute;
    bottom: 2px;
    right: -1px;

    @include media-breakpoint-up(md) {
      width: 15px;
      height: 15px;
      bottom: 8px;
      right: 7px;
    }
  }

  &__hidden_md_down {
    @include media-breakpoint-down(sm) {
      display: none;
    }
  }

  &__hidden_lg_up {
    @include media-breakpoint-up(md) {
      display: none;
    }
  }
}

.item_open {
  @include media-breakpoint-up(lg) {
    width: 53px;
    padding: 12px 6px;
  }
  &:last-child {
    margin-right: 0;
  }

  &__icon {
    width: 35px;
    height: 35px;
    fill: $orange;
  }

  &__badge {
    width: 15px;
    height: 15px;
    position: absolute;
    bottom: 2px;
    right: -1px;

    @include media-breakpoint-up(lg) {
      width: 15px;
      height: 15px;
      bottom: 8px;
      right: 7px;
    }
  }

  &__hidden_md_down {
    @include media-breakpoint-down(lg) {
      display: none;
    }
  }

  &__hidden_lg_up {
    @include media-breakpoint-up(xl) {
      display: none;
    }
  }
}

.item_menu {
  &:last-child {
    margin-right: 0;
  }

  &__icon {
    width: 35px;
    height: 35px;
    fill: $orange;
  }

  &__badge {
    width: 15px;
    height: 15px;
    position: absolute;
    bottom: 2px;
    right: -1px;

    @include media-breakpoint-up(md) {
      width: 15px;
      height: 15px;
      bottom: 8px;
      right: 7px;
    }
  }

  &__hidden_md_down {
    @include media-breakpoint-down(sm) {
      display: none;
    }
  }

  &__hidden_lg_up {
    @include media-breakpoint-up(md) {
      display: none;
    }
  }
}

.item_menu_open {
  &:last-child {
    margin-right: 0;
  }

  &__icon {
    width: 35px;
    height: 35px;
    fill: $orange;
  }

  &__badge {
    width: 15px;
    height: 15px;
    position: absolute;
    bottom: 5px;
    right: -5px;

    @include media-breakpoint-up(lg) {
      width: 15px;
      height: 15px;
      bottom: 8px;
      right: 7px;
    }
  }

  &__hidden_md_down {
    display: none;
    @include media-breakpoint-down(lg) {
      display: block;
    }
  }

  &__hidden_lg_up {
    @include media-breakpoint-up(xl) {
      display: none;
    }
  }
}

.hamburger_menu,
.hamburger_menu_open {
  position: absolute;
  display: none;
  align-items: center;
  justify-content: center;
  background-image: url(../../../../../../assets/images/footer/mobile-hamburger-menu.svg);
  flex-direction: column;
  top: -14rem;
  left: 0.5rem;
  background-size: 100%;
  background-position-y: bottom;
  padding: 0.5rem 0.5rem 1rem;
}

.hamburger_menu {
  @media screen and (max-width: 768px) {
    display: flex !important;
  }
  &__item {
    margin-right: 0;
  }
}

.hamburger_menu_open {
  @media screen and (max-width: 1200px) {
    display: flex !important;
  }
  &__item {
    margin-right: 0;
  }
}
