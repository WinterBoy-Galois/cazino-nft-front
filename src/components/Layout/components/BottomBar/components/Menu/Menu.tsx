import React, { useState, useEffect } from 'react';
import styles from './Menu.module.scss';
import Present from '../../../../../icons/Present';
import Badge from '../../../../../Badge';
import Book from '../../../../../icons/Book';
import VerifyLast from '../../../../../icons/VerifyLast';
import Hamburger from '../../../../../icons/Hamburger';
import Faucet from '../../../../../icons/Faucet';
import Sound from '../../../../../icons/Sound';
import { useStateValue } from '../../../../../../state';
import clsx from 'clsx';
import { useLocation, useNavigate } from '@reach/router';
import { useQuery } from '@apollo/client';
import { FAUCET_INFO } from '../../../../../../graphql/queries';
import { error as errorToast } from '../../../../../../components/Toast';
interface IProps {
  hasUnclaimedBonus?: boolean;
}

const Menu: React.FC<IProps> = ({ hasUnclaimedBonus }) => {
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const [isGamePage, setGamePage] = useState(false);
  const [{ auth }, dispatch] = useStateValue();
  const [isMenuOpened, setMenuOpened] = useState(false);
  const { refetch: refreshFaucetData } = useQuery(FAUCET_INFO);

  const [
    {
      sidebar: { isOpen, isSound },
    },
  ] = useStateValue();
  useEffect(() => {
    setGamePage(pathname.includes('/games'));
  }, [pathname]);
  useEffect(() => {
    setMenuOpened(false);
  }, [isOpen]);
  const onOpenFaucetModal = async () => {
    if (auth.state !== 'SIGNED_IN') {
      return await navigate(`${pathname}?dialog=sign-in`);
    }

    refreshFaucetData()
      .then(async response => {
        const { amount, canClaim, every, errors } = response.data.faucetInfo;

        if (errors) {
          if (errors[0].code === 'FAUCET_CLAIM_DISABLED') {
            return await navigate(`${pathname}?dialog=faucet`, {
              state: {
                errMessage: errors[0].message,
                timestamp: new Date(),
              },
            });
          } else {
            return errorToast(errors[0].message);
          }
        }

        return await navigate(`${pathname}?dialog=faucet`, {
          state: {
            amount: amount || 0,
            canClaim: canClaim || false,
            every: every || 0,
            timestamp: new Date(),
          },
        });
      })
      .catch(err => err);
  };

  const onClickSound = () => {
    dispatch({ type: 'SOUND_ON_OFF', payload: { isSound: !isSound } });
  };

  return (
    <div className={styles.container}>
      {isGamePage && isMenuOpened ? (
        <div className={clsx(isOpen ? styles.hamburger_menu_open : styles.hamburger_menu)}>
          <div
            className={clsx(
              isOpen ? styles.item_menu_open : styles.item_menu,
              styles.hamburger_menu__item
            )}
          >
            <Book className={clsx(isOpen ? styles.item_menu_open__icon : styles.item_menu__icon)} />
          </div>

          <div
            className={clsx(
              isOpen ? styles.item_menu_open : styles.item_menu,
              styles.hamburger_menu__item
            )}
          >
            <VerifyLast
              className={clsx(isOpen ? styles.item_menu_open__icon : styles.item_menu__icon)}
            />
          </div>

          <div
            className={clsx(
              isOpen ? styles.item_menu_open : styles.item_menu,
              styles.hamburger_menu__item
            )}
            onClick={onOpenFaucetModal}
          >
            <Faucet
              className={clsx(isOpen ? styles.item_menu_open__icon : styles.item_menu__icon)}
            />
          </div>

          <div
            className={clsx(
              isOpen ? styles.item_menu_open : styles.item_menu,
              styles.hamburger_menu__item
            )}
            onClick={onClickSound}
          >
            <Sound
              className={clsx(isOpen ? styles.item_menu_open__icon : styles.item_menu__icon)}
              sound={isSound}
            />
          </div>
        </div>
      ) : null}

      {isGamePage ? (
        <div
          className={clsx(
            isOpen
              ? clsx(styles.item_open, styles.item_open__hidden_lg_up)
              : clsx(styles.item, styles.item__hidden_lg_up)
          )}
          onClick={() => setMenuOpened(!isMenuOpened)}
        >
          <Hamburger
            className={clsx(isOpen ? styles.item_open__icon : styles.item__icon)}
            isOpened={isMenuOpened}
          />
        </div>
      ) : null}

      <div
        className={clsx(isOpen ? styles.item_open : styles.item)}
        onClick={() => dispatch({ type: 'MODAL_SHOW', payload: { type: 'USER_INFO_MODAL' } })}
      >
        <Present className={styles.item__icon} />
        {hasUnclaimedBonus ? (
          <div className={clsx(isOpen ? styles.item_open__badge : styles.item__badge)}>
            <Badge />
          </div>
        ) : null}
      </div>

      {isGamePage ? (
        <>
          <div
            className={clsx(
              isOpen
                ? clsx(styles.item_open, styles.item_open__hidden_md_down)
                : clsx(styles.item, styles.item__hidden_md_down)
            )}
          >
            <Book className={clsx(isOpen ? styles.item_open__icon : styles.item__icon)} />
          </div>

          <div
            className={clsx(
              isOpen
                ? clsx(styles.item_open, styles.item_open__hidden_md_down)
                : clsx(styles.item, styles.item__hidden_md_down)
            )}
          >
            <VerifyLast className={clsx(isOpen ? styles.item_open__icon : styles.item__icon)} />
          </div>

          <div
            className={clsx(
              isOpen
                ? clsx(styles.item_open, styles.item_open__hidden_md_down)
                : clsx(styles.item, styles.item__hidden_md_down)
            )}
            onClick={onOpenFaucetModal}
          >
            <Faucet className={clsx(isOpen ? styles.item_open__icon : styles.item__icon)} />
          </div>

          <div
            className={clsx(
              isOpen
                ? clsx(styles.item_open, styles.item_open__hidden_md_down)
                : clsx(styles.item, styles.item__hidden_md_down)
            )}
            onClick={onClickSound}
          >
            <Sound
              className={clsx(isOpen ? styles.item_open__icon : styles.item__icon)}
              sound={isSound}
            />
          </div>
        </>
      ) : null}
    </div>
  );
};

export default Menu;
