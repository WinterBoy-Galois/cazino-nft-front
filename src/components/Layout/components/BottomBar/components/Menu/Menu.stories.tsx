import React from 'react';
import { storiesOf } from '@storybook/react';
import Menu from '.';

storiesOf('Components/BottomBar/Menu', module)
  .add('default', () => <Menu />)
  .add('unclaimed', () => <Menu hasUnclaimedBonus={true} />);
