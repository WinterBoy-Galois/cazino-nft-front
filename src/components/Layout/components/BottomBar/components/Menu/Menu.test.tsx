import React from 'react';
import { render } from '@testing-library/react';
import Menu from './Menu';

describe('Menu', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<Menu />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
