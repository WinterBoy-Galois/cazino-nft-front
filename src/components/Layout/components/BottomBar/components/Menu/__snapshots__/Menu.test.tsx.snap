// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="item"
        >
          <svg
            class="item__icon"
            viewBox="0 0 34.45 34.81"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Слой 5"
                id="Слой_5"
              >
                <path
                  d="M34.43 16.07l-.49-3.34a1.77 1.77 0 00-1.76-1.53h-10v6.9h10.5a1.78 1.78 0 001.75-2.03zM.5 12.73L0 16.07a1.78 1.78 0 001.76 2h10.5V11.2h-10A1.78 1.78 0 00.5 12.73zM31.94 33V19.47h-9.77v15.34h7.9A1.87 1.87 0 0031.94 33zM2.51 19.47V33a1.86 1.86 0 001.86 1.86h7.91V19.47z"
                />
                <rect
                  height="23.61"
                  rx="1.29"
                  width="7.19"
                  x="13.63"
                  y="11.2"
                />
                <path
                  d="M26.47 5.66c-1.75-2.51-5.26.26-7 2.19a4 4 0 00-.67-.49C22.28 3.23 29.05-3.51 33 2.2c2.43 3.49-.38 7.4-11.67 8.11a3.76 3.76 0 00-.1-.66c4.77-.51 6.38-2.35 5.24-3.99z"
                />
                <path
                  d="M13 10.31a4.19 4.19 0 018.34 0z"
                />
                <path
                  d="M8 5.66c1.75-2.51 5.26.26 7 2.19a4 4 0 01.67-.49C12.16 3.23 5.38-3.51 1.4 2.2c-2.4 3.49.38 7.4 11.67 8.11a3.76 3.76 0 01.1-.66C8.45 9.14 6.83 7.3 8 5.66z"
                />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="item"
      >
        <svg
          class="item__icon"
          viewBox="0 0 34.45 34.81"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
            id="Layer_2"
          >
            <g
              data-name="Слой 5"
              id="Слой_5"
            >
              <path
                d="M34.43 16.07l-.49-3.34a1.77 1.77 0 00-1.76-1.53h-10v6.9h10.5a1.78 1.78 0 001.75-2.03zM.5 12.73L0 16.07a1.78 1.78 0 001.76 2h10.5V11.2h-10A1.78 1.78 0 00.5 12.73zM31.94 33V19.47h-9.77v15.34h7.9A1.87 1.87 0 0031.94 33zM2.51 19.47V33a1.86 1.86 0 001.86 1.86h7.91V19.47z"
              />
              <rect
                height="23.61"
                rx="1.29"
                width="7.19"
                x="13.63"
                y="11.2"
              />
              <path
                d="M26.47 5.66c-1.75-2.51-5.26.26-7 2.19a4 4 0 00-.67-.49C22.28 3.23 29.05-3.51 33 2.2c2.43 3.49-.38 7.4-11.67 8.11a3.76 3.76 0 00-.1-.66c4.77-.51 6.38-2.35 5.24-3.99z"
              />
              <path
                d="M13 10.31a4.19 4.19 0 018.34 0z"
              />
              <path
                d="M8 5.66c1.75-2.51 5.26.26 7 2.19a4 4 0 01.67-.49C12.16 3.23 5.38-3.51 1.4 2.2c-2.4 3.49.38 7.4 11.67 8.11a3.76 3.76 0 01.1-.66C8.45 9.14 6.83 7.3 8 5.66z"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
